transport/src/behaviour/base.rs

```diff
@@ -452,7 +452,7 @@ impl BaseBehaviour {
             #[cfg(feature = "metrics")]
             ONGOING_QUERIES.dec();
             return Some(ToSwarm::Dial {
-                opts: DialOpts::peer_id(peer_id).condition(PeerCondition::NotDialing).build(),
+                opts: DialOpts::peer_id(peer_id).condition(PeerCondition::NotDialing).build(), // Usually, the default of `DisconnectedAndNotDialing` is good enough unless you want to explicitly allow duplicated connections in case you already have one.
```

Comment:
I explicitly didn't want to use `DisconnectedAndNotDialing`, because sometimes we want to open a new *outbound* connection to a peer, even though an *inbound* connection already exists. It's part of our application logic to check if peer are dialable.